controller_interface/src/controller_interface.cpp

```diff
@@ -24,107 +24,17 @@
 
 namespace controller_interface
 {
-return_type ControllerInterface::init(
-  const std::string & controller_name, const std::string & namespace_,
-  const rclcpp::NodeOptions & node_options)
-{
-  node_ = std::make_shared<rclcpp_lifecycle::LifecycleNode>(
-    controller_name, namespace_, node_options, false);  // disable LifecycleNode service interfaces
-
-  try
-  {
-    auto_declare<int>("update_rate", 0);
-  }
-  catch (const std::exception & e)
-  {
-    fprintf(stderr, "Exception thrown during init stage with message: %s \n", e.what());
-    return return_type::ERROR;
-  }
-
-  switch (on_init())
-  {
-    case LifecycleNodeInterface::CallbackReturn::SUCCESS:
-      break;
-    case LifecycleNodeInterface::CallbackReturn::ERROR:
-    case LifecycleNodeInterface::CallbackReturn::FAILURE:
-      return return_type::ERROR;
-  }
-
-  node_->register_on_configure(
-    std::bind(&ControllerInterface::on_configure, this, std::placeholders::_1));
-
-  node_->register_on_cleanup(
-    std::bind(&ControllerInterface::on_cleanup, this, std::placeholders::_1));
-
-  node_->register_on_activate(
-    std::bind(&ControllerInterface::on_activate, this, std::placeholders::_1));
-
-  node_->register_on_deactivate(
-    std::bind(&ControllerInterface::on_deactivate, this, std::placeholders::_1));
+ControllerInterface::ControllerInterface() : ControllerInterfaceBase() {}
 
-  node_->register_on_shutdown(
-    std::bind(&ControllerInterface::on_shutdown, this, std::placeholders::_1));
+bool ControllerInterface::is_chainable() const { return false; }
 
-  node_->register_on_error(std::bind(&ControllerInterface::on_error, this, std::placeholders::_1));
-
-  return return_type::OK;
-}
-
-const rclcpp_lifecycle::State & ControllerInterface::configure()
+std::vector<hardware_interface::CommandInterface> ControllerInterface::export_reference_interfaces()
 {
-  // TODO(destogl): this should actually happen in "on_configure" but I am not sure how to get
-  // overrides correctly in combination with std::bind. The goal is to have the following calls:
-  // 1. CM: controller.get_node()->configure()
-  // 2. LifecycleNode: ControllerInterface::on_configure()
-  // 3. ControllerInterface: <controller>::on_configure()
-  // Then we don't need to do state-machine related checks.
-  //
-  // Other solution is to add check into the LifecycleNode if a transition is valid to trigger
-  if (get_state().id() == lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED)
-  {
-    update_rate_ = get_node()->get_parameter("update_rate").as_int();
-  }
-
-  return get_node()->configure();
+  return std::vector<hardware_interface::CommandInterface>();
```

Comment:
```suggestion
  return {};
```
how about something shorter? :)